portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,126 @@
+    
+    //response.setContentType("text/html;");
+    //response.getWriter().println("Hello Parth!");
+
+    //response.setContentType("application/json;");
+    //response.getWriter().println(json);
+    
+//   } 
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+    boolean sort = Boolean.parseBoolean(getParameter(request, "sort", "false"));
+
+    // Convert the text to upper case.
+    if (upperCase) {
+      text = text.toUpperCase();
+    }
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
+
+    // Sort the words.
+    if (sort) {
+      Arrays.sort(words);
+    }
+
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(Arrays.toString(words));
+
+    String input = request.getParameter("text-input");
+    String upper = request.getParameter("upper-case");
+    String sortInput = request.getParameter("sort");
+
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("text-input", input);
+    taskEntity.setProperty("upper-case", upper);
+    taskEntity.setProperty("sort", sortInput);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
   }
-}
+
+
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+  
+    private String convertToJson(ArrayList hello) {
+    String json = "{";
+    json += "\"firstWord\": ";
+    json += "\"" + hello.get(0) + "\"";
+    json += ", ";
+    json += "\"secondWord\": ";
+    json += "\"" + hello.get(1) + "\"";
+    json += ", ";
+    json += "\"thirdWord\": ";
+    json += hello.get(2);
+    json += ", ";
+    json += "\"fourthWord\": ";
+    json += hello.get(3);
+    json += "}";
+    return json;
+  }
+  
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Task").addSort("text-input", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<String> tasks = new ArrayList<String>();
+        for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String idNum = "" + id + "";
+        String textInput = (String) entity.getProperty("text-input");
+    
+        tasks.add(textInput);
+        tasks.add(idNum);
+    }
```

Comment:
[Style](https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation): Each time a new block or block-like construct is opened, the indent increases by two spaces